### ISSUE TITLE
Automatically refresh Token Identity for MFA authentication

### ISSUE THREAD

**Issue by PedroC88** [microsoft/azuredatastudio#16963]
When using MFA to connect to Azure, the token expires too soon (or doesn't refresh automatically). SSMS never looses the token even when the connection is closed, not until you fully close SSMS.

As a workaround I click on _Change Connection_ and it already comes preloaded with all the details of the current connection so it isn't that far away, however it would still be much better if it were refreshed automatically.

**Comment by Charles-Gagnon** [comment]
Closing as a duplicate of https://github.com/microsoft/azuredatastudio/issues/10352 - there's an open PR right now to fix this issue for query editors : https://github.com/microsoft/azuredatastudio/pull/16936
